core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2787,7 +2787,7 @@ private JoinBridgeManager<?> createLookupSourceFactoryManager(
 
             int taskConcurrency = getTaskConcurrency(session);
             OperatorFactory hashBuilderOperatorFactory;
-            if (useSpillingJoinOperator()) {
+            if (useSpillingJoinOperator(session)) {
                 hashBuilderOperatorFactory = new HashBuilderOperatorFactory(
                         buildContext.getNextOperatorId(),
                         node.getId(),
```

Comment:
Can `spillEnabled && partitionCount > 1` be replaced with `partitionCount > 1` ?

Reply:
No.
If `isForceSpillingOperator` is set but spilling is disabled this should return false, regardless of partitionCount